### PR TITLE
Sinanshamsudheen7 lok 186 billing implement credit based call initiation and deduction

### PR DESCRIPTION
**Pull request by sinanshamsudheen**
# What type of PR is this? (check all applicable)

- [ ] Refactor  
- [x] Feature  
- [ ] Improvement  
- [ ] Bug Fix  
- [ ] Optimization  
- [ ] Documentation Update  

## Description

This pull request implements a comprehensive credit-based system for call initiation and billing. Every new client signing up is granted 5 initial credits (equivalent to 5 minutes). The system checks if the user has sufficient credits before initiating any call and blocks initiation if credits are depleted or insufficient. After each completed call, the actual call duration (in minutes) is deducted from the user's remaining credits using data received through the Webhook analysis.

Key features implemented:
1. Credit allocation of 5 initial credits (5 minutes) on client signup
2. Credit checking mechanism before call initiation
3. Call blocking when credits are insufficient
4. Post-call credit deduction based on actual call duration
5. Integration with Webhook data for accurate billing
6. Appropriate error handling and user notifications when credits run out

## Screenshots, Recordings

_N/A - Backend feature implementation with no visual changes_

## QA Instructions

1. Create a new client account and verify 5 initial credits are allocated
2. Attempt to initiate a call with sufficient credits and verify it proceeds
3. Attempt to initiate a call with insufficient credits and verify it is blocked
4. Complete a call and verify credits are correctly deducted based on actual duration
5. Test edge cases like very short calls (ensure minimum billing)
6. Verify error messages are displayed appropriately when credits are insufficient
7. Test the webhook integration for accurate post-call credit adjustments

✅ Tested on **Firefox** on **Linux**

## I have tested the changes introduced in this pull request

- [x] Yes  
- [ ] No, and this is why: _please replace this line with details on why it wasn't possible_

## I have reviewed the changes first to ensure small things like console.logs and debug logic have been removed

- [x] Yes

## Added/updated tests?

_It is encouraged to keep the code coverage percentage at 70% and above._

- [x] Yes  
- [ ] No, and this is why: _Add reason if tests were not included_

### CODE REVIEW

File: server/app/core/config.py [hunk 1]
```diff
@@ -84,6 +84,15 @@ class Settings(BaseSettings):
...
     PUBLIC_API_PREFIX: str = "/api/v1/public"
     
+    # Billing and pricing settings
+    OPENAI_INPUT_COST_PER_1K: float = 0.005  # $5 per 1M input tokens
+    OPENAI_OUTPUT_COST_PER_1K: float = 0.020  # $20 per 1M output tokens
+    VAPI_COST_PER_MINUTE: float = 0.15  # 0.05 (base) + 0.10 (provider)
+    BILLING_MARGIN: float = 0.2  # Margin as float (0.2 = 20%)
+    
+    # Demo call billing setting
+    BILL_DEMO_CALLS: bool = False  # Whether to incur costs for demo calls, needed for testing (default: False)
```
Comment (raoofnaushad) [inline server/app/core/config.py:94]: Nice to have this. Thanks
